### DISCUSSION TITLE
Make GoKarts change their lane through the plugins API

### DISCUSSION THREAD

**Discussion by AdrianFahrbach** [OpenRCT2/OpenRCT2#18776] [Q&A]
Is there a way to make a go kart change its lane with a plugin?
The basic idea is to give the player the opportunity to change the lane of a cart during a race.
I've already tried by just changing their position, but that doesn't seem to work.

**Comment by spacek531** [comment]
If there's a hook for when a vehicle changes its track element, you might be able to manually change the subposition data type

**Reply by AdrianFahrbach** [reply]
I don't think that there is a hook yet.
I found parts like `VehicleTrackSubposition::GoKartsMovingToLeftLane` in the source code, but I have no idea of C++ nor the RCT code structure and therefore can't really tell what that does.

**Reply by spacek531** [reply]
What does AdvancedTrack use? That knows when a vehicle crosses a track element.

The different subpositions are basically different splines that the train can follow. Default spline is well default. Left lane is the spline that karts follow when they are in the left lane, Right lane is for the right lane. MovingToLeftLane is when the kart starts in the right lane and will move to the left lane by the end of the track element. MovingToRightLane is the opposite.
